File: src/main/java/komu/demodel/domain/Project.java
/*
 * Copyright (C) 2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.domain;

import static java.util.Collections.unmodifiableList;

import java.util.ArrayList;
import java.util.List;

import komu.demodel.utils.Check;

public final class Project {

    private final String name;
    private final List<InputSource> inputSources = new ArrayList<InputSource>();

    public Project(String name) {
        Check.notNull(name, "name");

        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void addInputSource(InputSource inputSource) {
        Check.notNull(inputSource, "inputSource");

        inputSources.add(inputSource);
    }

    public List<InputSource> getInputSources() {
        return unmodifiableList(inputSources);
    }
}

File: src/main/scala/komu/demodel/domain/project/Project.scala
/*
 * Copyright (C) 2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.domain.project

import java.util.Collections.unmodifiableList
import java.util.{ ArrayList, List }

final class Project(name: String) {

  val _inputSources = new ArrayList[InputSource]

  def addInputSource(inputSource: InputSource) {
    assert(inputSource != null)
    _inputSources.add(inputSource)
  }

  def inputSources = unmodifiableList(_inputSources)
}
